dotnet/Examples/ApproleResponseWrapping.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using VaultSharp;
+using VaultSharp.V1.AuthMethods;
+using VaultSharp.V1.AuthMethods.AppRole;
+using VaultSharp.V1.AuthMethods.Token;
+using VaultSharp.V1.Commons;
+
+namespace Examples 
+{
+    public class ApproleAuthExample
+    {
+        /// <summary>
+        /// Fetches a key-vaule secret (kv-v2) after authenticating to Vault via AppRole authentication
+        /// </summary>
+        public string GetSecretWithAppRole()
+        {
+            // A combination of a Role ID and Secret ID is required to log in to Vault with an AppRole.
+	        // The Secret ID is a value that needs to be protected, so instead of the app having knowledge of the secret ID directly,
+	        // we have a trusted orchestrator (https://learn.hashicorp.com/tutorials/vault/secure-introduction?in=vault/app-integration#trusted-orchestrator)
+	        // give the app access to a short-lived response-wrapping token (https://www.vaultproject.io/docs/concepts/response-wrapping).
+	        // Read more at: https://learn.hashicorp.com/tutorials/vault/approle-best-practices?in=vault/auth-methods#secretid-delivery-best-practices             
+            var roleId = Environment.GetEnvironmentVariable("APPROLE_ROLE_ID");
+            var vaultAddr = Environment.GetEnvironmentVariable("VAULT_ADDR");
+
+            string wrappingToken = File.ReadAllText("../../../path/to/wrapping-token"); // placed here by a trusted orchestrator
+
+            // We need to create two VaultClient objects for authenticating via AppRole. The first is for
+            // using the unwrap utility. We need to initalize the client with the wrapping token.
+            IAuthMethodInfo wrappedTokenAuthMethod = new TokenAuthMethodInfo(wrappingToken);
+            var wrappedVaultClientSettings = new VaultClientSettings(vaultAddr, wrappedTokenAuthMethod);
```

Comment:
It does make sense! I think it makes sense to rename the Vault Client, but possibly not the auth method, since the auth method is directly initialized with a wrapped token and that's how you are authenticating with Vault. What are your thoughts? b/c the auth method object isn't actually doing the unwrapping

Reply:
Yeah that makes sense, renaming the client and clientsettings variables but not the auth method. Sounds good to me!